c2corg_api/views/waypoint.py

```diff
@@ -32,16 +32,29 @@ def adapt_schema_for_type(waypoint_type):
     return restrict_schema(schema_waypoint, fields)
 
 
+@lru_cache(maxsize=None)
```

Comment:
It is hard to understand at first glance what `@lru_cache` does.
I think you should add a comment each time explaining what it is used for.